block/produce.go

```diff
@@ -228,3 +245,29 @@ func (m *Manager) createTMSignature(block *types.Block, proposerAddress []byte,
 	}
 	return vote.Signature, nil
 }
+
+// GetPreviousBlockHashes returns the hash of the last block and the commit for the last block
+// to be used as the previous block hash and commit for the next block
+func (m *Manager) GetPreviousBlockHashes(forHeight uint64) (lastHeaderHash [32]byte, lastCommit *types.Commit, err error) {
+	lastHeaderHash, lastCommit, err = loadPrevBlock(m.Store, forHeight-1)
+	if err != nil {
+		if !m.State.IsGenesis() { // allow prevBlock not to be found only on genesis
+			return [32]byte{}, nil, fmt.Errorf("load prev block: %w: %w", err, ErrNonRecoverable)
+		}
+		lastHeaderHash = [32]byte{}
```

Comment:
please define sentinel value instead of [32]byte{}

Reply:
acked